src/main/java/net/masterthought/cucumber/generators/AbstractPage.java

```diff
@@ -32,30 +12,20 @@
  */
 public abstract class AbstractPage {
 
-    private static final Logger LOG = LogManager.getLogger(AbstractPage.class);
-
-    private final VelocityEngine engine = new VelocityEngine();
-    protected final VelocityContext context = new VelocityContext();
-
     /** Name of the HTML file which will be generated. */
     private final String templateFileName;
-    /** Results of the report. */
-    protected final ReportResult reportResult;
-    /** Configuration used for this report execution. */
-    protected final Configuration configuration;
-
-    protected AbstractPage(ReportResult reportResult, String templateFileName, Configuration configuration) {
+    
+    protected AbstractPage(String templateFileName) {
```

Comment:
this changes public API but I see this is the price of this change...

Reply:
We could keep the previous public API deprecated.

Reply:
Good point, sounds like good compromise.